### DISCUSSION TITLE
Using array with more than five of elements in script on Tasmota 13.4.0 ESP32-C3 (Shelly Plus 1 Mini device)

### DISCUSSION THREAD

**Discussion by phidix** [arendst/Tasmota#20954] [Support request]
Hello,
I've compiled my firmware **Tasmota 13.4.0** for **ESP32-C3 (Shelly Plus 1 Mini device)** with the following flags:
```
#define USE_SCRIPT
#undef USE_RULES
#define USE_SCRIPT_SUB_COMMAND
#define USE_BUTTON_EVENT

#define USE_ESP32MAIL
#define SCRIPT_POWER_SECTION
#define LARGE_ARRAYS

#undef MAXFILT
#define MAXFILT 10
```

I'm trying to define and initialize permanent array with six of integer numbers in Tasmota script. Unfortunately for some reason every time I'm trying to set value of the element  with index=6 (or higher) its value is not set.
Below is the sample of the script's code with output:

```
>D
p:m:arr=0

>B
arr[1]=10
arr[2]=20
arr[3]=30
arr[4]=40
arr[5]=50
arr[6]=60

=>print count: %arr[-1]%
=>print 1: %arr[1]%
=>print 2: %arr[2]%
=>print 3: %arr[3]%
=>print 4: %arr[4]%
=>print 5: %arr[5]%
=>print 6: %arr[6]%
```

Below is the output:

> 18:01:48.802 RSL: SENSOR = {"Time":"2024-03-14T18:01:48","ANALOG":{"Temperature1":74.0},"TempUnit":"C"}
18:06:11.620 SCR: nv=1, tv=0, vns=4, vmem=61, smem=8192
18:06:11.622  count: 5.00
18:06:11.623  1: 10.00
18:06:11.624  2: 20.00
18:06:11.625  3: 30.00
18:06:11.626  4: 40.00
18:06:11.627  5: 50.00
18:06:11.628  6: 10.00

Do you know why this array is treated as the five-elements one / how to define the larger array?

Thanks in advance!

**Comment by sfromis** [comment]
Looks like it is not due to an array, but about using the median filter, which is always 5 entries.
> `m:vname`  specifies a median filter variable with 5 entries (for elimination of outliers)

**Reply by phidix** [reply]
Array definition changed to: 
`p:M:arr=0`
solved the issue :)

Thank you!
